### ISSUE TITLE
service tag regex should support '=' and ':'

### ISSUE THREAD

**Issue by Wing924** [hashicorp/consul-template#1049]
Because many system (ex. fabio) use '=' in service tags, we should support them in consul-template.

https://github.com/hashicorp/consul-template/blob/master/dependency/dependency.go#L21
AS IS:
```go
tagRe         = `((?P<tag>[[:word:]\.\-\_]+)\.)?`
```
TO BE:
```
tagRe         = `((?P<tag>[[:word:]=:\.\-\_]+)\.)?`
```

**Comment by pierresouchay** [comment]
@gmr The issue is that tags are supposed to be used within DNS as well, and those chars are out of scope of allowed chars.

IMHO Fabio might try switching to service.weights and service.Meta :-)

**Comment by gmr** [comment]
I just saw it as a request that the team asked for implementation on. If the scope should increase beyond the issue or request, I'm totally cool with closing it. Was just trying to help close out an issue that was tagged as wanting a contribution.

**Comment by pierresouchay** [comment]
@gmr Well, I don't think this is a bad idea however, I just explain why it is like this :-)

On my side in [consul-templaterb](https://github.com/criteo/consul-templaterb), I did not implement such limitation on chars within the tags, so I definitely it is a good idea :-)

**Comment by eikenb** [comment]
Merged in #1149 that adds this functionality.
